CMakeLists.txt

```diff
@@ -5,9 +5,9 @@ find_package(CUDA REQUIRED)
 find_package(Boost COMPONENTS program_options system filesystem REQUIRED)
 
 add_definitions("-std=c++1y")
-set(CUDA_NVCC_FLAGS_DEBUG ${CUDA_NVCC_FLAGS_DEBUG} -lineinfo -gencode arch=compute_35,code=sm_35 --ptxas-options -v )
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -rdynamic -Wall -DDEBUG -fsanitize=address")
-set(CUDA_NVCC_FLAGS_RELEASE ${CUDA_NVCC_FLAGS_RELEASE} -O2 -gencode arch=compute_35,code=sm_35 --ptxas-options -v)
+set(CUDA_NVCC_FLAGS_DEBUG ${CUDA_NVCC_FLAGS_DEBUG} -lineinfo -gencode arch=compute_30,code=sm_30 --ptxas-options -v )
```

Comment:
If we want to do it perfectly, it is better to add a cmake option to specify the `arch` and `code`.  But I agree that probably 30 is better than 35 and it should be the default.